lib/spack/spack/cmd/mirror.py

```diff
@@ -97,6 +98,19 @@ def setup_parser(subparser):
         default=spack.cmd.default_list_scope(),
         help="configuration scope to read from")
 
+    # Binary
```

Comment:
it's not a binary is it?  this is for source tarballs right?

Reply:
This particular PR is for the binary mirror  and while there is also a story for a source mirror, I mistakenly  moved that source mirror card into the _In Progress_ section, then referenced that card here in the PR message. Sorry about that. I presume this may have some consequence on the notes that you've left me here?

Reply:
I think binary caches are mostly similar, but they're handled by a different command.  Look at `spack buildcache create`, which takes specs of packages to create binary caches for, then puts them in a mirror directory structure.  It's the analog for `spack mirror create`.

I think for binaries, there should just be:

```
spack buildcache put <mirror> <spec>
```

And that should create the buildcache (if necessary) and upload it to the specified mirror.  The only piece of logic missing with the `buildcache` stuff is where to create the cache by default.  `spack buildcache create` currently assumes that `.` is a mirror if there is no `-d`, but I think it should probably put buildcaches in `var/spack/cache` (which is itself a mirror), similar to how tarballs are already put there.  Then this workflow would work:

```
spack mirror add production <s3 url>
spack buildcache put production hdf@1.4.5
```

And that should register an S3 bucket, then create and upload the `buildcache` to the mirror.

Seem reasonable?  The basic theme here is that the user shouldn't have to know too much about directory structure -- they just need to say what they want to do and name the points in configuration space (specs) that they want to deal with.